src/domain/cpu/workers/hello.worker.ts

```diff
@@ -0,0 +1,68 @@
+export interface DataInMessagePayload {
+  offset: number;
+  chunkLength: number;
+  pattern: Uint8Array;
+  text: Uint8Array;
+}
+
+export interface DataOutMessagePayload {
+  matches: Array<number>;
+}
+
+class SolverCPU {
+  private readonly ctx: Worker = (self as unknown) as Worker;
+
+  constructor() {
+    this.ctx.onmessage = this.handleMessage.bind(this);
+  }
+
+  handleMessage(message: MessageEvent<DataInMessagePayload>) {
+    if (message.data.pattern.length < 1 || message.data.text.length < 1) {
+      throw Error("Pattern and text must be at least 1 character long.");
+    }
+    if (message.data.pattern.length > message.data.text.length) {
+      throw Error("Pattern can't be longer than text.");
+    }
+    if (message.data.chunkLength < 1 || message.data.chunkLength > message.data.text.length){
+      throw Error("Chunk length must be greater than 0 and less than total text length.")
+    }
+    if (message.data.offset < 0 || message.data.chunkLength > message.data.text.length){
+      throw Error("Offset must be >= 0 and <= (text.length - pattern.length).")
+    }
```

Comment:
For clarity, I would check first for chunkLength < 1 and offset < 0 (in a one if statement), and then since offset and chunkLength are bound to each other I would check for their right limit with the condition:
`if (offset + pattern.length - 1 + chunkLength > text.length) {throw ...}`. I think that this should cover the wrong values.

Reply:
- Moved `DataInMessagePayload` validation to a separate method (`checkInputPayload`). 
- Improved the error messages to be more meaningful. 

However, I didn't implement the last expression check you mentioned. Isn't this check partially performed for "incomplete" chunks in the `solve` method?
```javascript
// Maximum valid index to start pattern matching from.
    const maxTextIdx = Math.min(
    data.offset + data.chunkLength - 1,
    data.text.length - data.pattern.length
 );
```

The expression you proposed would always fail for the last chunk, since we can't "overlap" matching positions onto the next chunk, since there is no next chunk.
I think that checking for reaching beyond the available text is required only for the first position in the chunk (offset). Every next pattern matching position would be handled determining `maxTextIdx` in the `solve` method.
I propose the following:
```javascript
if (data.offset + data.pattern.length > data.text.length) {
  throw Error(
    `Pattern matching (length ${data.pattern.length}) 
     at the offset (index ${data.offset}) would reach 
      beyond the available text (length ${data.text.length}).`
  );
}
```

Reply:
I pushed the proposed solution onto the branch. Please review the changes and if you agree - resolve this convo.

Reply:
Resolved in 1:1 VS Live Share session.